kratos/python_scripts/multistage_analysis.py

```diff
@@ -0,0 +1,149 @@
+        '''
+
+        if self.settings["stages"][self.GetCurrentStageName()].Has("stage_preprocess"):
+            if self.settings["stages"][self.GetCurrentStageName()]["stage_preprocess"].Has("modelers"):
+                for modeler in self.__GetModelers():
+                    modeler.SetupGeometryModel()
+                for modeler in self.__GetModelers():
+                    modeler.PrepareGeometryModel()
+                for modeler in self.__GetModelers():
+                    modeler.SetupModelPart()
+
+            if self.settings["stages"][self.GetCurrentStageName()]["stage_preprocess"].Has("operations"):
+                for operation in self.__GetOperations("stage_preprocess"):
+                    operation.Execute()
+
+    def RunCurrentStage(self):
+        '''This function executes (solves) the current stage.
+        Note that this call is equivalent to the traditional single-stage simulation run call.
+        '''
+
+        self.GetCurrentStage().Run()
+
+    def RunCurrentStagePostprocess(self):
+        '''This function executes the postprocessing of current stage.
+        Note that the stage postprocess deliberately involves operations only.
+        '''
+
+        if self.settings["stages"][self.GetCurrentStageName()].Has("stage_postprocess"):
+            if self.settings["stages"][self.GetCurrentStageName()]["stage_postprocess"].Has("operations"):
+                for operation in self.__GetOperations("stage_postprocess"):
+                    operation.Execute()
+
+    def GetNumberOfStages(self):
+        '''Returns the number of stages.'''
+
+        return len(self.__stages_map)
+
+    def GetCurrentStage(self):
+        '''Returns the current stage instance.'''
+
+        return self.__stages_map[self.__current_stage_name]
+
+    def GetCurrentStageName(self):
+        '''Returns the current stage name.'''
+        return self.__current_stage_name
+
+    def __GetExecutionList(self):
+        '''Creates and returns the execution list.
+        This method creates the execution list, either from a user-defined execution list or,
+        if this is not provided, from the stages declaration order.
+        '''
+
+        # Check if the execution list has been already created
+        if not hasattr(self, "__execution_list"):
+            # Default case in which the execution list is provided by the user
+            if self.settings.Has("execution_list"):
+                self.__execution_list = self.settings["execution_list"].GetStringArray()
+            # If not provided, create an auxiliary execution list from the stages declaration order
+            else:
+                KratosMultiphysics.Logger.PrintInfo("'execution_list' is not provided. Stages will be executed according to their declaration order.")
+                self.__execution_list = []
+                for stage_name in self.settings["stages"].keys():
+                    self.__execution_list.append(stage_name)
```

Comment:
```suggestion
                self.__execution_list = list(self.settings["stages"].keys())
```

The keys insertion order is only guranteed for python >= 3.7. In the case of python 3.6, only CPython implementations gurantee it. Since we are moving to python 3.8, i think this is ok. Just wanted to mention if we are gonna stay in python3.6 for little bit longer.

https://stackoverflow.com/questions/5629023/order-of-keys-in-dictionaries-in-old-versions-of-python

Reply:
As pointed by @ddiezrod, this is important for the @KratosMultiphysics/altair guys so another good reason for getting rid of the EOL Python 3.6.